monkey/infection_monkey/exploit/sshexec.py

```diff
@@ -73,15 +97,31 @@ def exploit_with_login_creds(self, port) -> paramiko.SSHClient:
             secrets=self.options["credentials"]["exploit_password_list"],
         )
 
-        for user, current_password in user_password_pairs:
+        credentials_iterator = interruptable_iter(
+            user_password_pairs,
+            self.interrupt,
+            "SSH exploiter has been interrupted",
+            logging.INFO,
+        )
+
+        for user, current_password in credentials_iterator:
 
             ssh = paramiko.SSHClient()
             ssh.set_missing_host_key_policy(paramiko.WarningPolicy())
             try:
-                ssh.connect(self.host.ip_addr, username=user, password=current_password, port=port)
+                ssh.connect(
+                    self.host.ip_addr,
+                    username=user,
+                    password=current_password,
+                    port=port,
+                    timeout=SSH_CONNECT_TIMEOUT,
+                    auth_timeout=SSH_AUTH_TIMEOUT,
+                    banner_timeout=SSH_BANNER_TIMEOUT,
+                )
 
                 logger.debug("Successfully logged in %r using SSH. User: %s", self.host, user)
                 self.add_vuln_port(port)
```

Comment:
Do we still need this call? If so, it's missing from `exploit_with_ssh_keys()`

Reply:
It is used in smb, wmi and msql. I will add it in `exploit with ssh_keys()`.